src/main/java/com/templateproject/api/controller/RecipeController.java

```diff
@@ -1,28 +1,141 @@
 package com.templateproject.api.controller;
 
+import com.templateproject.api.entity.IngredientRecipe;
 import com.templateproject.api.entity.Recipe;
 import com.templateproject.api.repository.RecipeRepository;
+import com.templateproject.api.repository.IngredientRecipeRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("/recipes")
 public class RecipeController {
-    @Autowired
-    RecipeRepository recipeRepository;
+final private RecipeRepository recipeRepository;
+final private IngredientRecipeRepository IngredientRecipeRepository;
 
+public RecipeController(RecipeRepository recipeRepository, IngredientRecipeRepository IngredientRecipeRepository) {
+        this.recipeRepository = recipeRepository;
+        this.IngredientRecipeRepository = IngredientRecipeRepository;
+    }
     // Get all recipes
+
     @GetMapping
     public List<Recipe> getAllRecipes() {
         return recipeRepository.findAll();
     }
 
+    @GetMapping("/{recipeName}")
+    public ResponseEntity<Recipe> searchRecipesByExactName(@PathVariable("recipeName") String recipeName) {
+        Recipe recipes = recipeRepository.findByRecipeNameIgnoreCase(recipeName);
+        return ResponseEntity.ok(recipes);
+    }
+
+    @GetMapping("/name")
+    public ResponseEntity<List<Recipe>> searchRecipesByPartialName(@RequestParam("recipeName") String recipeName) {
```

Comment:
some issues could occur if you're using the recipeName as pathVariable (accents, spaces, special characters) use a slug instead